pkg/apiserver/apic.go

```diff
@@ -0,0 +1,231 @@
+package apiserver
+
+import (
+	"context"
+	"net/url"
+	"sync"
+	"time"
+
+	"github.com/crowdsecurity/crowdsec/pkg/apiclient"
+	"github.com/crowdsecurity/crowdsec/pkg/apiserver/controllers"
+	"github.com/crowdsecurity/crowdsec/pkg/csconfig"
+	"github.com/crowdsecurity/crowdsec/pkg/database"
+	"github.com/crowdsecurity/crowdsec/pkg/models"
+	"github.com/crowdsecurity/crowdsec/pkg/types"
+	"github.com/go-openapi/strfmt"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+	"gopkg.in/tomb.v2"
+)
+
+const (
+	PullInterval    = "2m"
+	PushInterval    = "30s"
+	MetricsInterval = "30m"
+)
+
+type apic struct {
+	pullInterval    time.Duration
+	pushInterval    time.Duration
+	metricsInterval time.Duration
+	dbClient        *database.Client
+	apiClient       *apiclient.ApiClient
+	alertToPush     chan []*models.Alert
+	mu              sync.Mutex
+	pushTomb        tomb.Tomb
+	pullTomb        tomb.Tomb
+	metricsTomb     tomb.Tomb
+	startup         bool
+	credentials     *csconfig.ApiCredentialsCfg
+}
+
+func NewAPIC(config *csconfig.OnlineApiClientCfg, dbClient *database.Client) (*apic, error) {
+	var err error
+	var ret *apic
+
+	apiclient.BaseURL, err = url.Parse(config.Credentials.URL)
+	if err != nil {
+		return ret, errors.Wrapf(err, "parse local API URL '%s': %v ", config.Credentials.URL, err.Error())
+	}
+
+	password := strfmt.Password(config.Credentials.Password)
+	t := &apiclient.JWTTransport{
+		MachineID: &config.Credentials.Login,
+		Password:  &password,
+	}
+
+	pullInterval, err := time.ParseDuration(PullInterval)
```

Comment:
You don't want to make it part of OnlineApiClientCfg ? (same goes for pushInterval and MetricsInterval)